ddl/ddl_api.go

```diff
@@ -562,6 +566,19 @@ func columnDefToCol(ctx sessionctx.Context, offset int, colDef *ast.ColumnDef, o
 		Version: model.CurrLatestColumnInfoVersion,
 	})
 
+	if col.FieldType.Tp == mysql.TypeBlob {
+		if col.FieldType.Flen <= tinyBlobMaxLength {
+			logutil.BgLogger().Info("Auto convert Blob to TinyBlob", zap.Int("Blob Length", col.FieldType.Flen))
+			col.FieldType.Tp = mysql.TypeTinyBlob
+		} else if col.FieldType.Flen > blobMaxLength && col.FieldType.Flen <= mediumBlobMaxLength {
```

Comment:
```suggestion
		} else if col.FieldType.Flen > tinyBlobMaxLength && col.FieldType.Flen <= mediumBlobMaxLength {
```
BTW you don't need `if col.FieldType.Flen > blobMaxLength` actually here:

```
if col.FieldType.Flen <= tinyBlobMaxLength {
    ...
} else if col.FieldType.Flen <= mediumBlobMaxLength {
   ...
} else if ...
```